docs/adr/0001-daily-active-use-server-side-metrics-glean.md

```diff
@@ -0,0 +1,113 @@
+* B. StatsD and Grafana
+* C. Sql/Redash
+
+## Decision Outcome
+
+Chosen option:
+
+* A. "Glean for server-side measurement of DAU"
+
+The use of Glean appears the best choice for measuring internal DAU metrics.  It meets our requirements and provides us with needed support on the data processing side. It also provides considerable support from the Glean team to implement this in a thoughtful manner.  There are some challenges with this implementation (more below), namely in this being a greenfield attempt at Rust server-side metrics, however the pros outweigh the cons. Other metrics implementations like StatsD and Grafana cannot be easily used to measure and aggregate this data. Additionally, it adds considerable overhead in determining how to query the data and reconcile ping emissions of Sync events. Having dedicated organizational support means we establish best practices.
+
+## Pros and Cons of the Options
+
+### A. Glean
+
+Glean is a widely used tool at Mozilla and provides us with a solution to the given issue and possible extensibility in the future. Not without some challenges in initial implementation, but the potential for positive impact is high.
+
+#### Pros
+
+* Satisfaction of requirement to measure internal DAU metrics.
+* Core of Glean's purpose is to measure user interactions and the rich metadata that accompanies it.
+* Capacity for future expansion of application metrics within Sync beyond DAU.
+* Prepares for implementation of same measurements in autopush, also using Glean.
+* Easier to query.
+* Use of standardized Mozilla tooling.
+* Establishment of team knowledge of using Glean.
+* Establishment of server-side Rust best practices, leading to easier development for backend Rust applications.
+* Transparency in data review process with consideration made to minimum collection of data.
+
+#### Cons
+
+* The `glean_parser` tool used by other Glean applications is currently not supported for Rust. Furthermore, client applications can use the Glean SDK and this is also not an option for us.
+* Server side metrics have not yet been implemented for a Rust server application of this kind, so this is new territory.
+* There is added complexity of data review process and registration of the application to Glean's probe scraper.
+* Potential delays and challenges in new implementation.
+
+### StatsD and Grafana
+
+StatsD and Grafana offer us core application metrics and service health. While we use this frequently, it doesn't neatly fit the measurement requirements we have for DAU and would likely be very difficult to process via queries.
+
+#### Pros
+
+* Well understood and used.
+* Support for SRE for more complex queries.
+* Already utilized for core metrics.
+
+#### Cons
+
+* StatsD is not a good format for measuring something like user interactions.
+* Somewhat opaque and complicated query logic required.
+* Significant difficulty in aggregation and reconciliation logic. 
+* May not scale well given number of events.
```

Comment:
Might be good to say more about this: we haven't talked about the expected number of events and where this might hit a limit.